docs/src/modules/utils/useRouterExtra.ts

```diff
@@ -2,8 +2,9 @@ import { useRouter } from 'next/router';
 
 export default function useRouterExtra() {
   const router = useRouter();
-  const asPathWithoutLang = router.asPath.replace(/^\/[a-zA-Z]{2}\//, '/');
-  let product = asPathWithoutLang.replace(/^\/([^/]+)\/.*/, '$1');
+
+  const asPathWithoutLang = router.asPath.replace(/^\/+[a-zA-Z]{2}\//, '/');
```

Comment:
We usually do this in the rest of the codebase.

https://github.com/mui/material-ui/blob/d23b0fd33b7afdfb508ab171259130c682ba88d7/docs/src/modules/components/Demo.js#L58

https://github.com/mui/material-ui/blob/d23b0fd33b7afdfb508ab171259130c682ba88d7/docs/src/modules/components/Demo.js#L63

Would it make sense to do

```suggestion
  const { canonicalAs: asPathWithoutLang } = pathnameToLanguage(router.asPath);
```

And fix 

```jsx
const canonicalAs = userLanguage === 'en' ? pathname : pathname.substring(3);
```

?

Reply:
For this, I will open another PR to refactor the code and add more tests.